src/prefect/engine/flow_runner.py

```diff
@@ -395,6 +395,10 @@ def get_flow_run_state(
             for task in self.flow.sorted_tasks():
 
                 task_state = task_states.get(task)
+                if task_state is None and isinstance(
```

Comment:
This guard makes sense but also seems unnecessary -- is it to avoid the bad UX of supplying a specific state for a constant and not having the system respect it? (Even though that would be very unusual)

Reply:
@jlowin and I talked offline; going to merge this and also implement a large refactor for constants.